### ISSUE TITLE
[Bug] New version for google gemini  not support some old .env config

### ISSUE THREAD

**Issue by MuRo-J** [ChatGPTNextWeb/ChatGPT-Next-Web#3615]
**Describe the bug**

**Expected behavior**

1. The CUSTOM_MODELS env must not set
such as:
```
CUSTOM_MODELS=-all,+gemini-pro,+gpt-3.5-turbo,+gpt-4-1106-preview,gpt-4-1106-preview=gpt-4-turbo
```
get provider key error for home page 

 The HIDE_USER_API_KEY env also. must not set
```
HIDE_USER_API_KEY =1 
```
also get provider key error for home page.

2.
 The google key in .env set doesn't work 
but in the setting page is worked

sorry for now is to late hour ,I provide more information next day.

**Deployment**
 Server:  yarn build && yarn start

**Additional Logs**
Add any logs about the problem here.

**Comment by H0llyW00dzZ** [comment]
I know the issue where `The google key in .env set doesn't work` but I don't have access key for google gemini so can't fix/test for me
and here is the issue located need to fix
`auth.ts`
```ts
 // if user does not provide an api key, inject system api key
  if (!apiKey) {
    const serverApiKey = serverConfig.isAzure
      ? serverConfig.azureApiKey
      : serverConfig.apiKey;

    if (serverApiKey) {
      console.log("[Auth] use system api key");
      req.headers.set(
        "Authorization",
        `${serverConfig.isAzure ? "" : "Bearer "}${serverApiKey}`,
      );
    } else {
      console.log("[Auth] admin did not provide an api key");
    }
  } else {
    console.log("[Auth] use user api key");
  }

  return {
    error: false,
  };
}
```

**Comment by MuRo-J** [comment]
非常非常感谢您的回复, 对于 Access 问题现在可以很容易获取 Gemini API
https://deepmind.google/technologies/gemini/#build-with-gemini
访问后点击
![CleanShot 2023-12-25 at 01 16 51](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/64354330/0c45c36d-d3da-4f66-b908-b9e639634b2d)
然后j进入 https://ai.google.dev/   
继续get api key in google studio
![CleanShot 2023-12-25 at 01 19 34](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/64354330/77ec3b24-187a-4ff7-9729-5bf1d878bf73)

再点击 get api key 
![CleanShot 2023-12-25 at 01 18 11](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/64354330/bfb72940-e72e-4c7b-8c8f-2c29132139dd)
就可以获取啦，我也是为了测试是不是没有配置 Gemini相关 .evn导致无法进入首页 当时获取并尝试的，应该是开放获取啦


希望能帮上您～

**Comment by H0llyW00dzZ** [comment]
> 非常非常感谢您的回复, 对于 Access 问题现在可以很容易获取 Gemini API https://deepmind.google/technologies/gemini/#build-with-gemini 访问后点击 ![CleanShot 2023-12-25 at 01 16 51](https://private-user-images.githubusercontent.com/64354330/292682827-0c45c36d-d3da-4f66-b908-b9e639634b2d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM0Mzk4NDksIm5iZiI6MTcwMzQzOTU0OSwicGF0aCI6Ii82NDM1NDMzMC8yOTI2ODI4MjctMGM0NWMzNmQtZDNkYS00ZjY2LWI5MDgtYjllNjM5NjM0YjJkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI0VDE3MzkwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyZTFlODc0ZDYxYzI5MTlkZDFlNzIzMTY4MzAzY2JjZWIwOGIxN2NhOTY4ZTk2OWYxZDYzMzM3MTBkN2M5MjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.47HZYon_lQ6oOTwlZM-e2LboASMX561kohO5XAmeE1k) 然后j进入 https://ai.google.dev/ 继续get api key in google studio ![CleanShot 2023-12-25 at 01 19 34](https://private-user-images.githubusercontent.com/64354330/292682917-77ec3b24-187a-4ff7-9729-5bf1d878bf73.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM0Mzk4NDksIm5iZiI6MTcwMzQzOTU0OSwicGF0aCI6Ii82NDM1NDMzMC8yOTI2ODI5MTctNzdlYzNiMjQtMTg3YS00ZmY3LTk3MjktNWJmMWQ4NzhiZjczLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI0VDE3MzkwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU1NDQwN2JiMWUzNWFmMzk1YjNlMTgwMGRhZjM5YzliMTc2ZjBhMmI4YzYxNmI0ZTc3YWQyYjZkMWM4ZjZlYWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hGHTIda4EFdx-VikNLK4ILi1xmoaUT1YfBuWC6du3uw)
> 
> 再点击 get api key ![CleanShot 2023-12-25 at 01 18 11](https://private-user-images.githubusercontent.com/64354330/292682894-bfb72940-e72e-4c7b-8c8f-2c29132139dd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM0Mzk4NDksIm5iZiI6MTcwMzQzOTU0OSwicGF0aCI6Ii82NDM1NDMzMC8yOTI2ODI4OTQtYmZiNzI5NDAtZTcyZS00YzdiLThjOGYtMmMyOTEzMjEzOWRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI0VDE3MzkwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE1NGQ1MGIwY2UwZGU4NDYyNTYyNzA3YjgzZmFkYjVkMTMzNGU4Mzg3YmE5YzVjZTNiNDE3N2FkYzkyMWIwZmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.keJJUb-LykU6MsO2ahhZ8IFsV4BbEU8c9ArvizsPX0s) 就可以获取啦，我也是为了测试是不是没有配置 Gemini相关 .evn导致无法进入首页 当时获取并尝试的，应该是开放获取啦
> 
> 希望能帮上您～

check pull req
- Fix [Bug] Issue ChatGPTNextWeb#3615 #3617

that's should be fixed I guess, 
I write without test because don't have access but the code it's correct

**Comment by MuRo-J** [comment]
感谢处理，已经在改进了关闭这个Issue
